src/diagrams/common/common.ts

```diff
@@ -0,0 +1,147 @@
+import DOMPurify from 'dompurify';
+import { MermaidConfig } from '../../config.type';
+
+/**
+ * Gets the rows of lines in a string
+ *
+ * @param {string | undefined} s The string to check the lines for
+ * @returns {string[]} The rows in that string
+ */
+export const getRows = (s?: string): string[] => {
+  if (!s) return [''];
+  const str = breakToPlaceholder(s).replace(/\\n/g, '#br#');
+  return str.split('#br#');
+};
+
+/**
+ * Removes script tags from a text
+ *
+ * @param {string} txt The text to sanitize
+ * @returns {string} The safer text
+ */
+export const removeScript = (txt: string): string => {
+  return DOMPurify.sanitize(txt);
+};
+
+const sanitizeMore = (text: string, config: MermaidConfig) => {
+  // TODO Q: Should this check really be here? Feels like we should be sanitizing it regardless.
+  if (config.flowchart?.htmlLabels !== false) {
+    const level = config.securityLevel;
+    if (level === 'antiscript' || level === 'strict') {
+      text = removeScript(text);
+    } else if (level !== 'loose') {
+      text = breakToPlaceholder(text);
+      text = text.replace(/</g, '&lt;').replace(/>/g, '&gt;');
+      text = text.replace(/=/g, '&equals;');
+      text = placeholderToBreak(text);
+    }
+  }
+  return text;
+};
+
+export const sanitizeText = (text: string, config: MermaidConfig): string => {
+  if (!text) return text;
+  if (config.dompurifyConfig) {
+    text = DOMPurify.sanitize(sanitizeMore(text, config), config.dompurifyConfig).toString();
+  } else {
+    text = DOMPurify.sanitize(sanitizeMore(text, config));
+  }
+  return text;
+};
+
+export const sanitizeTextOrArray = (
+  a: string | string[] | string[][],
+  config: MermaidConfig
+): string | string[] => {
+  if (typeof a === 'string') return sanitizeText(a, config);
+  // TODO Q: Do we need flat?
+  return a.flat().map((x: string) => sanitizeText(x, config));
```

Comment:
@knsv

Reply:
a can be an array of arrays. Refactor, later.